semgrep/dependencyparser/find_lockfiles.py

```diff
@@ -0,0 +1,17 @@
+# find lockfiles
+import functools
+import os
+from pathlib import Path
+from typing import Generator
+
+TARGET_LOCKFILE_FILENAMES = ["pipfile.lock", "yarn.lock", "package-lock.json"]
+
+
+@functools.lru_cache(maxsize=None)
+def find_lockfiles(current_dir: Path) -> Generator[Path, None, None]:
+    for entry in os.scandir(current_dir):
+        full_path = Path(os.path.join(current_dir, entry.name))
+        if entry.is_dir():
+            yield from find_lockfiles(full_path)
```

Comment:
Do we need to worry about cycles cause of symlinks?

Reply:
@brendongo probably. I think I addressed in 89410c1